void_migration/void_migration.py

```diff
@@ -501,6 +698,19 @@ def time_march(p):
         if hasattr(p, "temperature"):
             T = thermal.update_temperature(s, T, p)  # delete the particles at the bottom of the hopper
 
+        if p.charge_discharge:
+            p = charge_discharge(p,op_arr,t)
+
+            p_count.append(np.count_nonzero(s[~np.isnan(s)]))
+
+            den = 1 - np.mean(np.isnan(s), axis=2)
+            
+            mass_in_fun.append(np.sum(mass_cal(den,s,p)))
+
+            if np.max(den) > 1.0:
+                print("*******************",np.max(den))
```

Comment:
can you remove all of these diagnostics from the main output? if you need these to be run some times, pull this out of the main function into a separate function, and add an optional `debug` flag that you can set from the params file.